lib/sim.c

```diff
@@ -907,12 +923,39 @@ sim_generate_parents(sim_t *self)
             goto out;
         }
     }
+    /* In case of a recombination event in the Kingman phase
+     * we require exactly one recombination point.
+     */
+    if(arg_flag == 1) {
+        node = C[0]->ancestry.head;
+        locus_mapping = (int_map_value_t *) node->item;
+        site_2 = (double)(locus_mapping->key);
+        node = node->next;
+        while(node != NULL) {
```

Comment:
Style nit agina --- ``while (node != NULL)`` . Otherwise, it looks like ``while`` is a function.